src/tools/device-graph/matter-device-graph.py

```diff
@@ -0,0 +1,197 @@
+#
+#    Copyright (c) 2022 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import os
+import pprint
+import sys
+
+import chip.clusters as Clusters
+import graphviz
+from rich.console import Console
+
+# Add the path to python_testing folder, in order to be able to import from matter_testing_support
+sys.path.append(os.path.abspath(sys.path[0] + "/../../python_testing"))
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main  # noqa: E402
+
+console = None
+
+
+# Given there is currently no tranlation from DeviceTypeID to the device type name,
+# this dict is created for now. When some more general is available, it should be updated to use this.
```

Comment:
The translation exists in our XML.  It's just the python codegen not using that part of the XML?

Reply:
I found a json file containing most of this in _examples/chef/sample_app_util/matter_device_types.json_, but that didn't seems like a file that should get pulled in to this tool. Is the XML you have in mind is src/app/zap-templates/zcl/data-model/chip/matter-devices.xml? I can try to incorporate that XML file into this tool instead of having this dict. 

Alternatively if anybody is aware of a way to do this translation from python, I'm all ears. I just haven't been able to find it.

Reply:
yeah, we just don't have this in the python codegen yet. At least not as far as I've found. It will end up in there from the spec DM-tools XML when we do the device type tests (not codegen'd, but available)

Reply:
Honestly, I'd just wait - let's settle on the DM tools as source of truth and clean this up as we go. Since this is part of the DM tools work (basically), we'll know to go back and consolidate.

Reply:
We could maybe hack a parser of `src/app/zap-templates/zcl/data-model/chip/matter-devices.xml` for now as a stand-alone method and replace that later with whatever the DM tools output.

We should switch for the DM tools once we validate they are complete and bug-free. For now matter-devices is what zap uses, so it probably is reasonably complete.

Reply:
When we write a parser for that though, we should strip out the `Matter` or `Ma-` smurf-style naming of everything.

Reply:
I don't think that's a blocker for this PR though - seems orthogonal to getting the graph-generation bits in.

Reply:
As soon as we have this in a "better" format, I will certainly make sure to update this tool to use this. I have no intention of maintaining this list in the long term 😄